gem/interpreter.py

```diff
@@ -172,6 +172,28 @@ def _evaluate_operator(e, self):
     return result
 
 
+@_evaluate.register(gem.MathFunction)  # noqa: F811
+def _(e, self):
+    ops = [self(o) for o in e.children]
+    result = Result.empty(*ops)
+    names = {"abs": abs,
+             "log": math.log}
+    op = names[e.name]
+    if e.name is 'imag':
+        for idx in numpy.ndindex(result.tshape):
+            result[idx] = [o[o.filter(idx,result.fids)].imag for o in ops]
+    elif e.name is 'conj':
+        for idx in numpy.ndindex(result.tshape):
+            result[idx] = [o[o.filter(idx,result.fids)].conjugate() for o in ops]
+    elif e.name is 'real':
+        for idx in numpy.ndindex(result.tshape):
+            result[idx] = [o[o.filter(idx,result.fids)].real for o in ops]
+    else:
+        for idx in numpy.ndindex(result.tshape):
+            result[idx] = op(*(o[o.filter(idx, result.fids)] for o in ops))
```

Comment:
So this is not performance critical? It seems to be done the slow way rather than the fast way (see `_evaluate_operator`).

Reply:
Also it's ugly that the loop is duplicated several times with very tiny changes to each.

Reply:
AKA:
```
ops = [self(o) for o in e.children]
result = Result.empty(*ops)
op = {"abs": numpy.ndarray.abs,
                "log": numpy.ndarray.log,
                "imag": operator.attrgetter("imag"),
                "real": operator.attrgetter("real"),
                "conf": numpy.ndarray.conjugate}[e.name]
fids = result.fids
result.arr = op(*[o.broadcast(fids) for o in ops])
return result
```

I think.

Reply:
Maybe `numpy.concatenate([op(o.broadcast(fids)) for o in ops])`